app/models/provider.rb

```diff
@@ -1,5 +1,10 @@
 class Provider < ApplicationRecord
-  devise :saml_authenticatable, :trackable
+  encrypts :auth_subject_uid, deterministic: true
+
+  # devise :saml_authenticatable, :trackable
+  # devise :trackable, :omniauthable, omniauth_providers: [:entra_id]
```

Comment:
must be removaed due to mutlipke oauth providers against different user types: https://github.com/heartcombo/devise/wiki/OmniAuth-with-multiple-models